### PR TITLE
feature(nemesis): add disrupt_truncate_large_partition

### PR DESCRIPTION
**Pull request by amoskong**
Trello ticket: https://trello.com/c/rMSCxzit/2168-abort-compactions-more-frequently

Introduced a new truncate nemesis, it truncates a large-partition table,
it's used to cover one improvement of compaction.
The increase frequency of checking abortion is very useful for truncate.

Signed-off-by: Amos Kong <amos@scylladb.com>

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by bentsi** [comment]
@amoskong please address comments

**Comment by amoskong** [comment]
> @amoskong please address comments

I got another task to bisect a kernel bug, so the PR issues haven't addressed.
Will update the PR within two days.

File: sdcm/nemesis.py [hunk 1]
```diff
@@ -780,6 +780,27 @@ def disrupt_truncate(self):
...
+        """
+        self._set_current_disruption('TruncateMonkey {}'.format(self.target_node))
+
+        keyspace_truncate = 'scylla_bench'
+        table = 'test'
+        stress_cmd = "scylla-bench -workload=sequential -mode=write -replication-factor=3 -partition-count=100 " + \
+                     "-clustering-row-count=5555 -clustering-row-size=uniform:1024..2048 -concurrency=10 " + \
+                     "-connection-count=10 -consistency-level=quorum -rows-per-request=10 -timeout=30s -validate-data"
+        cs_thread = self.tester.run_stress_thread(stress_cmd=stress_cmd, keyspace_name=keyspace_truncate, stop_test_on_failure=False)
+        cs_thread.verify_results()
```
[the remark this answers is not in the document]
Comment (amoskong) [reply sdcm/nemesis.py:797]: It's a special keyspace, I'm using 'scylla_bench'.

Maybe keyspace_truncate should be renamed to 'truncated_keyspace'

[the remark this answers is not in the document]
Comment (amoskong) [reply sdcm/nemesis.py:797]: Maybe I should rename keyspace_truncate to ks_name
And set it to 'ks_truncate_large_partition'.

[the remark this answers is not in the document]
Comment (amoskong) [reply sdcm/nemesis.py:803]: keyspace_name parameter isn't supported for loader.run_stress_thread_bench().
I will add '--keyspace ks_truncate_large_partition --table test_table' to scylla-bench cmdline inside this nemesis.

File: sdcm/nemesis.py [hunk 2]
```diff
@@ -786,6 +786,29 @@ def disrupt_truncate(self):
         # do the actual truncation
         self.target_node.run_cqlsh(cmd='TRUNCATE {}.{}'.format(keyspace_truncate, table), timeout=120)
 
+    def disrupt_truncate_large_partition(self):
+        """
+        Introduced a new truncate nemesis, it truncates a large-partition table,
+        it's used to cover one improvement of compaction.
+        The increase frequency of checking abortion is very useful for truncate.
+        """
+        self._set_current_disruption('TruncateMonkey {}'.format(self.target_node))
```
Comment (roydahan) [inline sdcm/nemesis.py:795]: Change the name to something else, we already have TruncateMonkey

**Pull request merged**